main.go

```diff
@@ -88,29 +84,32 @@ func main() {
 	if err := validateHostPort(config.DnsAddr); err != nil {
 		log.Fatalf("addr is invalid: %s\n", err)
 	}
-	config, err := loadConfig(client, config)
+
+	config, err := server.LoadConfig(client, config)
 	if err != nil {
 		log.Fatal(err)
 	}
-	s := NewServer(config, client)
-	if s.config.Local != "" {
-		s.config.Local = dns.Fqdn(s.config.Local)
+	if config.Local != "" {
+		config.Local = dns.Fqdn(config.Local)
 	}
+	s := server.New(config, client)
 
 	if discover {
 		go func() {
 			recv := make(chan *etcd.Response)
-			go s.client.Watch("/_etcd/machines/", 0, true, recv, nil)
+			go client.Watch("/_etcd/machines/", 0, true, recv, nil)
 			duration := 1 * time.Second
 			for {
 				select {
 				case n := <-recv:
 					if n != nil {
-						s.UpdateClient(n)
+						if client := updateClient(n, tlskey, tlspem, cacert); client != nil {
+							s.UpdateClient(client)
+						}
 						duration = 1 * time.Second // reset
 					} else {
 						// we can see an n == nil, probably when we can't connect to etcd.
-						s.config.log.Infof("ectd machine cluster update failed, sleeping %s + ~3s", duration)
+						log.Printf("ectd machine cluster update failed, sleeping %s + ~3s", duration)
```

Comment:
Because it's in main (so tls static vars don't have to be part of server), it doesn't have access to the config.log.  The other alternative was to move all the tls config into Config so that UpdateClient could correctly create a tls client, but update client was only called from master.  If you prefer the latter I can put it into its own goroutine to loop.


> On Jan 16, 2015, at 2:57 AM, Miek Gieben <notifications@github.com> wrote:
> 
> In main.go:
> 
> >  						duration = 1 * time.Second // reset
> >  					} else {
> >  						// we can see an n == nil, probably when we can't connect to etcd.
> > -						s.config.log.Infof("ectd machine cluster update failed, sleeping %s + ~3s", duration)
> > +						log.Printf("ectd machine cluster update failed, sleeping %s + ~3s", duration)
> This changes, changes the output as well, as any reason not to use the Infof here?
> 
> —
> Reply to this email directly or view it on GitHub.
>